sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.cpp

```diff
@@ -607,8 +607,14 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMGetMemAllocInfo(
 }
 
 static ur_result_t USMFreeImpl(ur_context_handle_t Context, void *Ptr) {
-  ZE2UR_CALL(zeMemFree, (Context->ZeContext, Ptr));
-  return UR_RESULT_SUCCESS;
+  auto ZeResult = ZE_CALL_NOCHECK(zeMemFree, (Context->ZeContext, Ptr));
+  // Handle When the driver is already released
+  if (ZeResult == ZE_RESULT_ERROR_UNINITIALIZED ||
+      ZeResult == ZE_RESULT_SUCCESS) {
```

Comment:
The `ZE_RESULT_SUCCESS` should be properly handled by `ze2urResult`:
```suggestion
  if (ZeResult == ZE_RESULT_ERROR_UNINITIALIZED) {
```

Reply:
fixed in latest patch.